File: create-math-docx.js
const { Document, Paragraph, TextRun, Math, Packer } = require('docx');
const fs = require('fs');

// Create a new document
const doc = new Document({
    sections: [
        {
            properties: {},
            children: [
                new Paragraph({
                    children: [
                        new TextRun("Single variable calculus formulas: "),
                        new Math("A = \u03C0r^2"),
                        new TextRun(" (area of a circle)"),
                    ],
                }),
                new Paragraph({
                    children: [
                        new TextRun("Derivative of sin(x): "),
                        new Math("\u2202/\u2202x sin(x) = cos(x)"),
                    ],
                }),
                new Paragraph({
                    children: [
                        new TextRun("Quadratic formula: "),
                        new Math("x = \u00B1 \u221A(b^2 - 4ac) / 2a"),
                    ],
                }),
                new Paragraph({
                    children: [
                        new TextRun("Integral of x^n: "),
                        new Math("\u222B x^n dx = x^(n+1)/(n+1) + C"),
                    ],
                }),
            ],
        },
    ],
});

// Save the document
docx.Packer.toBuffer(doc).then((buffer) => {
    fs.writeFileSync('math-formulas.docx', buffer);
    console.log('Document saved successfully!');
});

File: create-math-docx.py
from docx import Document
from docx.oxml.ns import nsdecls
from docx.oxml import parse_xml

# Create a new document
doc = Document()

# Add a paragraph with a math formula
p = doc.add_paragraph('The formula for the area of a circle is: ')

# Add OMML math formula (A = πr²)
mathml = parse_xml('''
<w:r xmlns:w="http://schemas.openxmlformats.org/wordprocessingml/2006/main">
    <w:rPr><w:noProof/></w:rPr>
    <m:oMath xmlns:m="http://schemas.openxmlformats.org/officeDocument/2006/math">
        <m:r>
            <m:t>A</m:t>
        </m:r>
        <m:r>
            <m:t>=</m:t>
        </m:r>
        <m:r>
            <m:t>π</m:t>
        </m:r>
        <m:r>
            <m:t>r</m:t>
        </m:r>
        <m:sup>
            <m:r>
                <m:t>2</m:t>
            </m:r>
        </m:sup>
    </m:oMath>
</w:r>
''')

p._element.append(mathml)

# Save the document
doc.save('math-formula.docx')

print('Document saved as math-formula.docx')
